src/main/java/org/folio/service/AcquisitionsUnitsService.java

```diff
@@ -81,8 +81,9 @@ public Future<AcquisitionsUnit> getAcquisitionsUnit(String id, RequestContext re
   }
 
   public Future<Void> deleteAcquisitionsUnit(String id, RequestContext requestContext) {
-    return getAcquisitionsUnit(id, requestContext).map(unit -> unit.withIsDeleted(true))
-      .compose(unit -> updateAcquisitionsUnit(unit, requestContext));
+    // propose actully delete the unit instead of change it's isDelete to true
+    RequestEntry requestEntry = new RequestEntry(ENDPOINT_ACQ_UNITS_BY_ID).withId(id);
+    return restClient.delete(requestEntry, requestContext);
```

Comment:
As an option to resolve this you can modify this method and together with setting is_deleted flag to true you can update acq unit to some random name, for example using this formula: previousName_randomString. As length of random string you can choose for example 5 symbols.

Reply:
yep, I agree. then, should not unit names be unique? since they are stored in invoice, orders etc. duplicate name will cause confuse, right? or we should be have buttons like "pause the unit" and "recover the unit" if users want to use the same name.

Reply:
yep, will do